core/src/main/java/nl/pim16aap2/bigDoors/moveBlocks/BlockMover.java

```diff
@@ -73,20 +73,24 @@ private void preprocess()
                 continue;
 
             final Block block = newLocation.getBlock();
-            if (block.getPistonMoveReaction() == PistonMoveReaction.BREAK)
-                Bukkit.getScheduler().scheduleSyncDelayedTask(plugin, block::breakNaturally, 1L);
+            BigDoors.getScheduler().runTask(block.getLocation(), () -> {
+                if (block.getPistonMoveReaction() == PistonMoveReaction.BREAK)
+                    BigDoors.getScheduler().runTaskLater(block.getLocation(), block::breakNaturally, 1L);
+            });
```

Comment:
Why is the if-statement put inside another runnable?

Reply:
getPistonMoveReaction() needs to be run on the region thread or it errors